lib/srv/discovery/fetchers/aws-sync/s3.go

```diff
@@ -65,8 +66,14 @@ func (a *awsFetcher) fetchS3Buckets(ctx context.Context) ([]*accessgraphv1alpha.
 		}
 	}
 
+	region := awsutil.GetKnownRegions()[0]
+	if len(a.Regions) > 0 {
+		region = a.Regions[0]
+	}
```

Comment:
are we picking a "random" region for this API call? will this overwrite AWS_REGION from env? (concern is whether this will break AWS CN or US gov regions)

Reply:
The call replaces the `region` with the user's set regions `a.Regions[0]`.
Since US and CN regions are different partitions, it will inherit them from those settings.

Reply:
my bad, I read the condition opposite!